tpm2/update.go

```diff
@@ -32,6 +32,23 @@ import (
 
 var skdbUpdatePCRProtectionPolicyNoValidate = (*sealedKeyDataBase).updatePCRProtectionPolicyNoValidate
 
+// pcrPolicyVersionOption describes how to determine the version of a new PCR policy.
+type pcrPolicyVersionOption int
+
+const (
+	// resetPcrPolicyVersion indicates that the new policy version should be set
+	// to the current counter value.
+	resetPcrPolicyVersion pcrPolicyVersionOption = iota
+
+	// newPcrPolicyVersion indicates that the new policy version should be the current
+	// counter value plus 1.
+	newPcrPolicyVersion
+
+	// incrementPolicyVersion indicates that the new policy version should be the
+	// previous policy version plus 1.
+	incrementPcrPolicyVersion
```

Comment:
I got slightly confused by this and how it differs from the `newPcrPolicyVersion` option. Should we expand its comment a bit to indicate that this is used only by legacy/v0,v1,v2 TPM keydata versions? (I believe the rest of the difference in logic can be inferred by following the corresponding flows and reading other doc comments)

Reply:
I agree that some further clarification about the 3 modes would be good